src/Hl7.Fhir.Core/Model/ModelInfo.cs

```diff
@@ -473,6 +474,24 @@ public static string CanonicalUriForFhirCoreType(FHIRAllTypes type)
             return CanonicalUriForFhirCoreType(type.GetLiteral());
         }
 
+        // The lazy here is not used for purposes of performance (delaying creation of the versioned provider, which
+        // is cheap anyway), but to use the framework to have a nicely multithread-aware singleton.
+        private static readonly Lazy<ModelInspector> _versionedProvider
+            = new Lazy<ModelInspector>(createVersionedProvider);
+
+        private static ModelInspector createVersionedProvider()
+        {
+            var result = new ModelInspector(Specification.FhirRelease.STU3);
+
+            result.Import(typeof(Resource).GetTypeInfo().Assembly);
```

Comment:
In fact you want to import from `Hl7.Fhir.Support.Poco` and `Hl7.Fhir.Core`. Should we not be more specific about this then? When Patient moves to Common, this code will not work for example..

Reply:
Yes......using ModelInfo is obviously better.